### PR TITLE
fix: pop with dialog context

### PR DESCRIPTION
**Pull request by theachoem**
Dialog use its provided context to pop. This cause the context to be disposed and can't be reused.

```dart
var result = await showAdaptiveDialog(context)

// cause error here as context is disposed.
if(ok) showSnackBar(context)
```

### CODE REVIEW

File: lib/src/alert_dialog/show_alert_dialog.dart [hunk 1]
```diff
@@ -30,11 +30,13 @@ Future<T?> showAlertDialog<T>({
...
-    context,
-    rootNavigator: useRootNavigator,
-  );
-  void pop(T? key) => navigator.pop(key);
+  NavigatorState navigator(BuildContext context) {
+    return Navigator.of(
+      context,
+      rootNavigator: useRootNavigator,
+    );
+  }
```
Comment (mono0926) [inline lib/src/alert_dialog/show_alert_dialog.dart:38]: I think this is better.

```dart
  void pop({required BuildContext context, required T? key}) => Navigator.of(
        context,
        rootNavigator: useRootNavigator,
      ).pop(key);
```

```dart
onPressed: (key) => pop(context: context, key: key),
```

(I just realized that there was no point in separating the navigator and pop methods originally.)

**Review by mono0926 — COMMENTED** [review]
@theachoem 

Thanks for the PR.

I think this is better 🤔

File: lib/src/alert_dialog/show_alert_dialog.dart [hunk 2]
```diff
@@ -30,11 +30,6 @@ Future<T?> showAlertDialog<T>({
   Widget? macOSApplicationIcon,
   RouteSettings? routeSettings,
 }) {
-  final navigator = Navigator.of(
-    context,
-    rootNavigator: useRootNavigator,
-  );
-  void pop(T? key) => navigator.pop(key);
```
Comment (mono0926) [inline lib/src/alert_dialog/show_alert_dialog.dart:37]:
```dart
void pop({required BuildContext context, required T? key}) => Navigator.of(
        context,
        rootNavigator: useRootNavigator,
      ).pop(key);
```

File: lib/src/alert_dialog/show_alert_dialog.dart [hunk 3]
```diff
@@ -75,7 +71,7 @@ Future<T?> showAlertDialog<T>({
               actions: actions
                   .map(
                     (a) => a.convertToIOSDialogAction(
-                      onPressed: pop,
+                      onPressed: (key) => Navigator.of(context, rootNavigator: useRootNavigator).pop(key),
```
Comment (mono0926) [inline lib/src/alert_dialog/show_alert_dialog.dart:74]:
```suggestion
                      onPressed: (key) => pop(context: context, key: key),
```

so are the others.

**Review by mono0926 — COMMENTED** [review]
https://github.com/mono0926/adaptive_dialog/pull/115/commits/7eaed50927f0df2a934addc3c587ddaf4d310782 didn't follow my review( https://github.com/mono0926/adaptive_dialog/pull/115#discussion_r1231979152 ).

File: lib/src/alert_dialog/show_alert_dialog.dart (hunk 3 above)
Comment (theachoem) [reply lib/src/alert_dialog/show_alert_dialog.dart:74]: Let me fix that

**Comment by mono0926** [comment]
@theachoem 

Thanks, released: https://github.com/mono0926/adaptive_dialog/releases/tag/1.9.0-no-macos.3

**Pull request merged**